cpmpy/expressions/core.py

```diff
@@ -107,7 +107,29 @@ def __init__(self, name, arg_list):
                 arg_list[i] = arg_list[i].reshape(-1)
 
         assert (is_any_list(arg_list)), "_list_ of arguments required, even if of length one e.g. [arg]"
-        self.args = arg_list
+        self._args = arg_list
+
+
+    @property
+    def args(self):
+        return self._args
+
+    @property
+    def vars(self):
```

Comment:
vars() returns _args?

Reply:
Ah I forgot some parts that use this, this is the default value. Only get's overwritten in tables for now, because there are a lot of constants in tables. makes get_variables a lot faster. Maybe we can do it for weighted sums as well